internal/host/host.go

```diff
@@ -877,15 +898,20 @@ func (m *Manager) selectRole(ctx context.Context, h *models.Host, db *gorm.DB) (
 
 	// count already existing masters
 	mastersCount := 0
-	if err := db.Model(&models.Host{}).Where("cluster_id = ? and status != ? and role = ?",
+	if err = db.Model(&models.Host{}).Where("cluster_id = ? and status != ? and role = ?",
 		h.ClusterID, models.HostStatusDisabled, models.HostRoleMaster).Count(&mastersCount).Error; err != nil {
 		log.WithError(err).Errorf("failed to count masters in cluster %s", h.ClusterID.String())
 		return autoSelectedRole, err
 	}
 
 	if mastersCount < common.MinMasterHostsNeededForInstallation {
 		h.Role = models.HostRoleMaster
-		vc, err := newValidationContext(h, db, m.hwValidator)
+		clusterFromDB, err = common.GetClusterFromDBWithoutDisabledHosts(db, h.ClusterID)
```

Comment:
maybe instead of performing the get, `newValidationContext` should handle it? 
If cluster is nil then it can take it  from DB. 
Because it's a pointer it will be safer as well.

Reply:
OK